internal/engine/configs/api.go

```diff
@@ -205,6 +207,32 @@ func toCmdObjects(tlr tiltfile.TiltfileLoadResult) typedObjectSet {
 	return result
 }
 
+// Pulls out all the UIResource objects generated by the Tiltfile.
+func toUIResourceObjects(tlr tiltfile.TiltfileLoadResult) typedObjectSet {
+	result := typedObjectSet{}
+
+	for _, m := range tlr.Manifests {
+		name := string(m.Name)
+		labels := make(map[string]string)
+		for _, l := range m.Labels {
```

Comment:
it seems odd for the tiltfile to use two different data types as the source and the destination of labels - seems likely to lead to mistakes

Reply:
do you have a suggestion for how to handle the source type vs the destination type? the source type is only relevant in the context of the tiltfile, but we do the same data transformation in the `webview` and `api`.

Reply:
i mean that the tiltfile tracks labels as []string, but they're ultimately sent as map[string]string. Let's just use a map everywhere

Reply:
gotcha! i paired with matt to move the mapping logic and typing into the starlark unpacking func, so there aren't two data types for labels. does that address your concern?